components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java

```diff
@@ -200,6 +206,40 @@ public UserStoreResponse editUserStore(String domainId, UserStoreReq userStoreRe
         }
     }
 
+    /**
+     * Retrieve primary user store.
+     *
+     * @return UserStoreConfigurationsRes.
+     */
+    public UserStoreConfigurationsRes getPrimaryUserStore() {
+
+        RealmService realmService = UserStoreConfigServiceHolder.getInstance().getRealmService();
+        RealmConfiguration realmConfiguration = realmService.getBootstrapRealmConfiguration();
+        if (realmConfiguration == null) {
+            throw handleException(Response.Status.INTERNAL_SERVER_ERROR, UserStoreConstants.ErrorMessage.
+                    ERROR_CODE_ERROR_RETRIEVING_PRIMARY_USERSTORE);
+        }
+        List<AddUserStorePropertiesRes> propertiesTobeAdd = new ArrayList<>();
+        UserStoreConfigurationsRes userStoreConfigurations = new UserStoreConfigurationsRes();
+        userStoreConfigurations.setClassName(realmConfiguration.getUserStoreClass());
+        userStoreConfigurations.setDescription(realmConfiguration.getDescription());
+        userStoreConfigurations.setName(UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME);
+        userStoreConfigurations.setTypeId(base64URLEncodeId(Objects.requireNonNull
+                (getUserStoreTypeName(realmConfiguration.getUserStoreClass()))));
+        userStoreConfigurations.setTypeName(getUserStoreTypeName(realmConfiguration.getUserStoreClass()));
+        Map<String, String> userstoreProps = realmConfiguration.getUserStoreProperties();
+        if (MapUtils.isNotEmpty(userstoreProps)) {
+            for (String propKey : userstoreProps.keySet()) {
+                AddUserStorePropertiesRes userStorePropertiesRes = new AddUserStorePropertiesRes();
+                userStorePropertiesRes.setName(propKey);
+                userStorePropertiesRes.setValue(userstoreProps.get(propKey));
+                propertiesTobeAdd.add(userStorePropertiesRes);
+            }
+        }
+        userStoreConfigurations.setProperties(propertiesTobeAdd);
+        return userStoreConfigurations;
```

Comment:
As the primary user store, config does not change during the run time shall we keep this in a static variable? build on in that variable is null?